core/src/main/java/org/infinispan/statetransfer/StateTransferInterceptor.java

```diff
@@ -357,7 +361,7 @@ private Object handleTxWriteReturn(InvocationContext rCtx, VisitableCommand rCom
          throws Throwable {
       int retryTopologyId = -1;
       WriteCommand writeCommand = (WriteCommand) rCommand;
-      if (t instanceof OutdatedTopologyException) {
+      if (t instanceof OutdatedTopologyException || t instanceof AllOwnersLostException) {
```

Comment:
You mean in JGroupsTransport? There it's a different level of abstraction... I would decide probably after your glorious withdrawal from `MessageDispatcher`.